### PR TITLE
Pre stage to stage

### PR DESCRIPTION
**Pull request by axelrod-blox**

### CODE REVIEW

**Review by IlyaVi — APPROVED** [review]

File: src/services/keyShare.service.ts [hunk 1]
```diff
@@ -105,7 +107,7 @@ export const getResponse = (keyShareResponseId: KeyShareValidationResponseId, er
   }
 };
 
-export const getTooltipText = (count: number) => translations.VALIDATOR.BULK_REGISTRATION.SELECTED_VALIDATORS_TOOLTIP(count);
+export const getTooltipText = (count: number, condition: boolean): string | false => condition && translations.VALIDATOR.BULK_REGISTRATION.SELECTED_VALIDATORS_TOOLTIP(count);
```
Comment (alexstolr) [inline src/services/keyShare.service.ts:110]: I think you should have the move the false outside of getTooltipText to where getTooltipText is called.

File: src/app/components/applications/SSV/ValidatorRegistrationConfirmation/ValidatorRegistrationConfirmation.tsx [hunk 2]
```diff
@@ -52,7 +52,7 @@ const ValidatorRegistrationConfirmation = () => {
   if (Number(liquidationCollateralCost) < ssvStore.minimumLiquidationCollateral) {
     liquidationCollateralCost = new Decimal(ssvStore.minimumLiquidationCollateral);
   }
-  const amountOfSsv = formatNumberToUi(liquidationCollateralCost.add(networkCost).add(operatorsCost).toString());
+  const amountOfSsv = formatNumberToUi(liquidationCollateralCost.add(networkCost).add(operatorsCost).mul(validatorStore.validatorsCount).toString());
```
Comment (alexstolr) [inline src/app/components/applications/SSV/ValidatorRegistrationConfirmation/ValidatorRegistrationConfirmation.tsx:55]: I think toString() can be removed

File: src/app/components/applications/SSV/RegisterValidatorHome/components/OfflineKeyShareGeneration/OfflineKeyShareGeneration.tsx [hunk 3]
```diff
@@ -83,7 +83,7 @@ const OfflineKeyShareGeneration = () => {
   };
 
   const goToChangeOperators = () => {
-    navigate(-4);
+    navigate(-2);
```
Comment (alexstolr) [inline src/app/components/applications/SSV/RegisterValidatorHome/components/OfflineKeyShareGeneration/OfflineKeyShareGeneration.tsx:86]: What is the meaning of -2 when navigating?

File: src/app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/KeyShares.tsx [hunk 4]
```diff
@@ -211,7 +216,7 @@ const KeyShareFlow = () => {
         setWarningMessage(warningTextMessage);
         setValidatorsCount(Object.values(validators).filter((validator: ValidatorType) => validator.isSelected).length);
         setSelectedOperatorsData(operatorsData.sort((operatorA: SelectedOperatorData, operatorB: SelectedOperatorData) => Number(operatorA.operatorId) - Number(operatorB.operatorId)));
-        setMaxAvailableValidatorsCount(maxValidatorsCount);
+        setMaxAvailableValidatorsCount(Object.values(validators).filter((validator: ValidatorType) => validator.isSelected).length < maxValidatorsCount ? Object.values(validators).filter((validator: ValidatorType) => validator.isSelected).length : maxValidatorsCount);
```
Comment (alexstolr) [inline src/app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/KeyShares.tsx:219]: can you make Object.values(validators).filter((validator: ValidatorType) => validator.isSelected).length a const? you are calculating it twice.

**Review by alexstolr — APPROVED** [review]
Please see comments

File: src/app/components/applications/SSV/RegisterValidatorHome/components/OfflineKeyShareGeneration/OfflineKeyShareGeneration.tsx (hunk 3 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/RegisterValidatorHome/components/OfflineKeyShareGeneration/OfflineKeyShareGeneration.tsx:86]: routing -2 steps return us to choose operators

File: src/services/keyShare.service.ts (hunk 1 above)
Comment (axelrod-blox) [reply src/services/keyShare.service.ts:110]: u mean move the condition outside?

File: src/app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/KeyShares.tsx (hunk 4 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/KeyShares.tsx:219]: +

File: src/app/components/applications/SSV/ValidatorRegistrationConfirmation/ValidatorRegistrationConfirmation.tsx (hunk 2 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/ValidatorRegistrationConfirmation/ValidatorRegistrationConfirmation.tsx:55]: +

File: src/app/components/applications/SSV/RegisterValidatorHome/components/FundingPeriod/FundingPeriod.tsx [hunk 5]
```diff
@@ -55,7 +55,7 @@ const FundingPeriod = () => {
 
   const getDisableStateCondition = () => {
     if (isCustomPayment) {
-      return customPeriod <= 0 || isNaN(customPeriod);
+      return new Decimal(customPeriod).comparedTo(ssvStore.walletSsvBalance) === 1 || isNaN(customPeriod);
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/RegisterValidatorHome/components/FundingPeriod/FundingPeriod.tsx:58]: refactored

**Pull request merged**